pygen/pygen_src/riscv_reg.py

```diff
@@ -0,0 +1,131 @@
+"""
+Copyright 2020 Google LLC
+Copyright 2020 PerfectVIPs Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+"""
+
+import sys
+import logging
+import vsc
+from importlib import import_module
+from pygen_src.riscv_instr_gen_config import cfg
+from pygen_src.riscv_instr_pkg import (
+    pkg_ins, privileged_reg_t, reg_field_access_t, privileged_level_t)
+rcs = import_module("pygen_src.target." + cfg.argv.target + ".riscv_core_setting")
+
+
+# Light weight RISC-V register class library
+# Base class for RISC-V register field
+@vsc.randobj
+class riscv_reg_field:
+    def __init__(self):
+        self.bit_width = 0
+        self.reset_val = vsc.bit_t(rcs.XLEN)
+        self.val = vsc.rand_bit_t(rcs.XLEN)
+        self.access_type = vsc.enum_t(reg_field_access_t)
+        self.hard_wired = vsc.bit_t(1)
+        self.name = ""
+
+    @vsc.constraint
+    def zero_reserved_field_c(self):
+        with vsc.implies(self.access_type == reg_field_access_t.WPRI):
+            self.val == 0
+
+    @vsc.constraint
+    def hardwired_fld_c(self):
+        with vsc.implies(self.hard_wired == 1):
+            self.val == self.reset_val
+
+    def covert2string(self):
+        return pkg_ins.format_string("{} bit_width:{} val:{} type:{}".format(
+            self, self.bit_width, hex(self.val), self.access_type))
+
+    def post_randomize(self):
+        mask = vsc.bit_t(rcs.XLEN)
+        mask = 1
```

Comment:
this value should actually be `XLEN'1`, instead of 1

Reply:
I think this will still show it as XLEN'00..001, but I suggest combine the two lines: mask = vcs.bit_t(rcs.XLEN, 1). WDYT?

Reply:
whoops sorry, I meant `'1` (in systemverilog land), which would translate to `0xffff_..._ffff

Reply:
Ahh got it.